tests/queries/0_stateless/00911_tautological_compare.sql

```diff
@@ -1,10 +1,10 @@
-SELECT count() FROM system.numbers WHERE number != number;
-SELECT count() FROM system.numbers WHERE number < number;
-SELECT count() FROM system.numbers WHERE number > number;
+-- SELECT count() FROM system.numbers WHERE number != number;
```

Comment:
I hope we will revert this?

Reply:
It is not so easy to do.
It's actually not so clear why reading from infinite table **must** return at all.
We can go farter and want a query `SELECT count() FROM system.numbers WHERE number < 10` return `10`. But is it necessary?

This optimizations is implemented in a not obvious way for now. We expected that a function `!=` for two same non-constant arguments will return constant `0`, and assume that constant will remain. It could have been ok if we had some meta info that columns are really same. But we just compare two ptrs. I don't think there is a guarantee that it is true for all other blocks. (However, I could not find counterexample. Queries like `where x != if(blockNumber() == 0, x, y)` work correctly only because `blockNumber` does not return constant.)